Products/GenericSetup/registry.py

```diff
@@ -644,6 +644,10 @@ def __init__(self):
     def getProfileInfo(self, profile_id, for_=None):
         """ See IProfileRegistry.
         """
+        if profile_id.startswith("profile-"):
+            profile_id = profile_id[len('profile-'):]
```

Comment:
You have the same code repeatedly, which smells of possible typos. Tests will determine whether it's working correctly and is likely to flag up errors precisely because of the fragility.

The alternative would be to use a regex for `'profile-|snapshot-'`